driver-core/src/main/com/mongodb/ConnectionString.java

```diff
@@ -486,12 +498,20 @@ public ConnectionString(final String connectionString) {
 
 
     private void warnOnUnsupportedOptions(final Map<String, List<String>> optionsMap) {
-        for (final String key : optionsMap.keySet()) {
-            if (!ALL_KEYS.contains(key)) {
-                if (LOGGER.isWarnEnabled()) {
-                    LOGGER.warn(format("Connection string contains unsupported option '%s'.", key));
-                }
-            }
+        if (LOGGER.isWarnEnabled()) {
+            optionsMap.keySet()
+                    .stream()
+                    .filter(k -> !ALL_KEYS.contains(k))
+                    .forEach(k -> LOGGER.warn(format("Connection string contains unsupported option '%s'.", k)));
+        }
+    }
+
+    private void warnDeprecatedTimeouts(final Map<String, List<String>> optionsMap) {
```

Comment:
I opted to warn users if they are using deprecated timeouts in the connection string.

Reply:
Actual deprecations in the builders for these timeouts will occur in a future PR.